### ISSUE TITLE
r/networking_subnet_v2 allocation_pool - possibly docs?

### ISSUE THREAD

**Issue by keiranmraine** [terraform-providers/terraform-provider-openstack#764]
### Terraform Version

```bash
$ terraform -v
Terraform v0.12.1
+ provider.openstack v1.19.0
```

### Affected Resource(s)
- r/networking_subnet_v2 -> allocation_pool

### Expected Behavior
`allocation_pool` should accept an array according to [docs](https://www.terraform.io/docs/providers/openstack/r/networking_subnet_v2.html#allocation_pool).

### Actual Behavior
Validate gives errors if anything other than a **block** is defined, it is not clear what the structure should be.

### Steps to Reproduce

This validates but doesn't appear to align with the documentation:

> allocation_pool - (Optional) An array of sub-ranges of CIDR available for dynamic allocation to ports. The allocation_pool object structure is documented below.

```bash
$ echo 'provider "openstack" {
  version = "~> 1.19"
}
resource "openstack_networking_network_v2" "os_network" {
  name = "test_network"
}
resource "openstack_networking_subnet_v2" "os_subnet" {
  name            = "test_subnet"
  network_id      = "${openstack_networking_network_v2.os_network.id}"
  ip_version      = 4
  cidr            = "192.168.254.0/24"
  dns_nameservers = ["172.18.255.1", "172.18.255.2", "172.18.255.3"]
  allocation_pool {
      start = "192.168.254.2"
      end   = "192.168.254.200"
  }
}' > test.tf
$ terraform init
$ terraform validate
```

### References
Outcome of #729, #739.

**Comment by jtopjian** [comment]
@keiranmraine Thank you for reporting this. This indeed looks to be a bad description of `allocation_pool` in the docs.

I've opened #765 to remedy this. Please let me know if the new description makes sense.

**Comment by keiranmraine** [comment]
@jtopjian I've added a couple of comments to #765.

**Comment by jtopjian** [comment]
Fixed in #765
